src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/EnumMutatorFactory.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.jazzer.mutation.mutator.lang;
+
+import static com.code_intelligence.jazzer.mutation.combinator.MutatorCombinators.mutateIntegralInRange;
+import static com.code_intelligence.jazzer.mutation.combinator.MutatorCombinators.mutateThenMapToImmutable;
+import static com.code_intelligence.jazzer.mutation.support.TypeSupport.findFirstParentIfClass;
+
+import com.code_intelligence.jazzer.mutation.api.MutatorFactory;
+import com.code_intelligence.jazzer.mutation.api.SerializingMutator;
+import java.lang.reflect.AnnotatedType;
+import java.util.Optional;
+
+final class EnumMutatorFactory extends MutatorFactory {
+  @Override
+  public Optional<SerializingMutator<?>> tryCreate(AnnotatedType type, MutatorFactory factory) {
+    return findFirstParentIfClass(type, Enum.class).map(parent -> {
+      Enum[] values = ((Class<Enum>) type.getType()).getEnumConstants();
+      // TODO: when 1 or 0 values, return a constant mutator?
+      if (values.length == 0) {
+        return null;
```

Comment:
The equivalent of `null` for `Optional<T>` is `Optional.empty()`. While `null` is a valid return value, it shouldn't be used together with `Optional`s.

Reply:
Do we need Optional.empty()  here? At the moment we require at least 2 elements in the enum.

Reply:
As far as I can tell we would now reach the `require` in `mutateIndices` if an `Enum` has less than two elements. We could have a more user-facing error message by requiring right in the `tryCreate` function with a message such as "%s defines less than two enum constants and can't be mutated, use a constant instead".